src/dbnode/storage/shard.go

```diff
@@ -726,6 +745,7 @@ func (s *dbShard) tickAndExpire(
 	// future read lock attempts.
 	blockStates := s.blockStatesSnapshotWithRLock()
 	s.RUnlock()
+	// NB: no lock is held when the func is called. See the implementation of forEachShardEntryBatch().
```

Comment:
That is nice, looks like there is a snapshot lock above, what is the impact if we hold that lock for longer time.

Reply:
Are you referring to `blockStates := s.blockStatesSnapshotWithRLock()`? No, that function call doesn't try to hold a lock. The name means the callsite should hold `s.RLock()` while calling `blockStatesSnapshotWithRLock`.